code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -228,7 +228,17 @@ public void show(final boolean withMessagingDelegateControl) {
                                 // Show the MessageFragment with iam.
                                 final FragmentManager fragmentManager =
                                         currentActivity.getFragmentManager();
-                                messageFragment.show(fragmentManager, FRAGMENT_TAG);
+                                try {
+                                    messageFragment.show(fragmentManager, FRAGMENT_TAG);
```

Comment:
Based on the stack trace 
```
  java.lang.IllegalStateException: Activity has been destroyed
        at android.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1913)
        at android.app.BackStackRecord.commitInternal(BackStackRecord.java:688)
        at android.app.BackStackRecord.commit(BackStackRecord.java:646)
        at android.app.DialogFragment.show(DialogFragment.java:241)
        at com.adobe.marketing.mobile.services.ui.AEPMessage.lambda$show$0$com-adobe-marketing-mobile-services-ui-AEPMessage(AEPMessage.java:231)
```

I suspect that the following in happening : 
1. `AEPMessage.show()` is called 
     1a. `ActivityA` is fetched in Line#189 of this class to be the "currentActivity"
     1b. ExecutorService from AndroidUIService queues this runnable for execution. 
2. By the time the runnable 1b executes ActivityA is no longer the current activity (likely stopped) and say `ActivityB` is now the current activity.
3. L#220 in this class attempts to run the `messageFragment.show()` on `ActivityA`'s reference resulting in the illegal state exception.

Did you happen to try this scenario for a repro? Fetching the current activity within the runnable ie Line#217 will provide the most recent activity and perhaps mitigate the IllegalStateException that we are seeing here while increasing the chances of showing the message.
To prevent a crash, we can then wrap this in a try/catch for the edge case scenarios. Thoughts?